### ISSUE TITLE
[clang] compiles malformed code successfully: variant with array

### ISSUE THREAD

**Issue by tryfinally** [llvm/llvm-project#104049]
clang upto trunk compiles a variant with an array alternative. 
https://godbolt.org/z/Gf3M69v4K

MSVC correctly rejects the code.

std::variant requires all alternative types to be Cpp17Destructible. See  [22.6.3.1](https://eel.is/c++draft/variant#variant.general-2)
arrays are not [Cpp17Destructible](https://eel.is/c++draft/utility.arg.requirements#:Cpp17Destructible)

```cpp
#include <variant>

std::variant<int[3]> v;

int f() {
    return std::get<0>(v)[1];
}
```

**Comment by AaronBallman** [comment]
This is not an issue with Clang or with libc++, it's a libstdc++ issue and should be reported at https://gcc.gnu.org/bugzilla/

https://godbolt.org/z/8v1jc3b33 shows Clang's behavior with libc++ and GCC's behavior by default.
